Test/PcapDeviceTest.cs

```diff
@@ -51,12 +60,82 @@ public void DeviceProperties([PcapDevices] DeviceFixture fixture)
             Assert.IsNotNull(pcapIf.GatewayAddresses);
             Assert.IsNotNull(pcapIf.Addresses);
 
+            var resolution = device.TimestampPrecision;
+            if(targetResolution != null)
+            {
+                // confirm the resolution was set
+                if(targetResolution != resolution)
+                {
+                    Console.WriteLine("Error: {0} expected {1} but was {2}", device.Name, targetResolution, resolution);
+                } else
+                {
+                    Console.WriteLine("Pass: {0} expected {1} but was {2}", device.Name, targetResolution, resolution);
+                }
+
+                // Note: This doesn't not appear to work for any platform so it is
+                // disabled for now
+                //                Assert.AreEqual(targetResolution, resolution);
+            }
+
+            Assert.IsNotNull(device.TimestampsSupported);
+
             if (pcapIf.MacAddress != null)
             {
                 Assert.That(pcapIf.MacAddress.GetAddressBytes(), Has.Length.EqualTo(6));
             }
         }
 
+        /// <summary>
+        /// Note: This tests the default timestamp precision as set by libpcap internally
+        /// </summary>
+        /// <param name="fixture"></param>
+        [Test]
+        public void DeviceOpen([PcapDevices] DeviceFixture fixture)
+        {
+            using var device = (PcapDevice)fixture.GetDevice();
+
+            device.Open();
+            DeviceTest(device, null);
+        }
+
+        [Test]
+        public void DeviceOpenWithTimestampPrecision([PcapDevices] DeviceFixture fixture)
+        {
+            using var device = (PcapDevice)fixture.GetDevice();
+
+            foreach (TimestampResolution resolution in Enum.GetValues(typeof(TimestampResolution)))
+            {
+                var configuration = new DeviceConfiguration();
+                configuration.TimestampResolution = resolution;
+                device.Open(configuration);
+                DeviceTest(device, resolution);
```

Comment:
this won't run for all timestamps, since you don't close the device at the end of each iteration